primary/tests/causal_completion_tests.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::time::Duration;
+use test_utils::cluster::Cluster;
+use tracing::{info, subscriber::set_global_default};
+use tracing_subscriber::filter::{EnvFilter, LevelFilter};
+
+#[ignore]
+#[tokio::test]
+async fn test_read_causal_signed_certificates() {
+    const CURRENT_ROUND_METRIC: &str = "narwhal_primary_current_round";
+
+    // Enabled debug tracing so we can easily observe the
+    // nodes logs.
+    setup_tracing();
+
+    let mut cluster = Cluster::new(None);
+
+    // start the cluster
+    let nodes = cluster.start(4).await;
+
+    // Let primaries advance little bit
+    tokio::time::sleep(Duration::from_secs(10)).await;
+
+    // Ensure all nodes advanced
+    for node in nodes {
```

Comment:
I believe at the moment the metrics is the most powerful tool we have to understand what's going on with a node since we are lacking of any endpoints - at least to test the cases that have to do with node advancement, late joiners etc